tests/Integration/Tool/WebPageTextGetterTest.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Unit\LLPhant\Tool;
+
+use LLPhant\Tool\WebPageTextGetter;
+
+it('tests a full embedding flow with Redis', function () {
+    $url = 'https://en.wikipedia.org/wiki/Chemical_element';
```

Comment:
The page will be updated so you should save. We don't need the whole page maybe juste a sub part is enough?

+ the goal of the tool is to give the result to an llm so maybe remove line break when there are more than 2. And remove all JS ;)